docs/src/prototypes/ChatPeoplePicker/ChatPeoplePicker.tsx

```diff
@@ -0,0 +1,360 @@
+  editText: {
+    input: {
+      width: '100%',
+    },
+    variables: {
+      inputFocusBorderColor: 'transparent',
+    },
+    div: {
+      flexBasis: '100px',
+      flexGrow: 1,
+    },
+  },
+  listboxUL: {
+    position: 'absolute',
+    zIndex: 1000,
+  },
+  ariaLive: {
+    border: '0px',
+    clip: 'rect(0px, 0px, 0px, 0px)',
+    height: '1px',
+    margin: '-1px',
+    overflow: 'hidden',
+    padding: '0px',
+    width: '1px',
+    position: 'absolute',
+  },
+}
+
+interface IPeoplePickerProps {
+  source: (
+    inputValue: string,
+    selected: any[],
+  ) => { name: string; image: string; position: string }[]
+  styles?: any
+}
+
+interface IPeoplePickerState {
+  selected: any[]
+  focused: boolean
+  inputValue: string
+  deleteOnBackspace: boolean
+  available: any[]
+  message: string
+}
+
+export class ChatPeoplePicker extends React.Component<IPeoplePickerProps, IPeoplePickerState> {
+  private input: any
+
+  constructor(props) {
+    super(props, {})
+
+    this.state = {
+      selected: [],
+      focused: false,
+      inputValue: '',
+      deleteOnBackspace: true,
+      available: this.props.source('', []),
+      message: '',
+    }
+
+    this.input = React.createRef()
+  }
+
+  private handleCloseIconAction(element, event) {
+    this.removeFromSelected(element)
+    this.input.current.inputRef.focus()
+    event.stopPropagation()
+  }
+
+  private addToSelected(element) {
+    this.setState(({ selected }) => {
+      const newSelected = [...selected, element]
+      return {
+        selected: newSelected,
+        available: this.props.source('', newSelected),
+        inputValue: '',
+        message: `${element.name} has been selected.`,
+      }
+    })
+  }
+
+  private removeFromSelected(element?) {
+    let { selected } = this.state
+    let poppedElement
+    const { inputValue } = this.state
+    if (element) {
+      selected = selected.filter(currentElement => currentElement !== element)
+    } else {
+      poppedElement = selected.pop()
+    }
+    this.setState({
+      selected,
+      available: this.props.source(inputValue, selected),
+      message: `${(element || poppedElement).name} has been removed.`,
+    })
+  }
+
+  public render(): React.ReactNode {
+    return (
+      <div style={this.props.styles}>
+        {/* Added label outside because otherwise I could click outside the combobox on the same line with the label and listbox would not close. */}
+        <Downshift
+          stateReducer={this.stateReducer}
+          onChange={this.onDropdownChange}
+          selectedItem={null}
+          inputValue={this.state.inputValue}
+          itemToString={item => (item ? item.value : '')}
+        >
+          {({
+            getInputProps,
+            getItemProps,
+            getMenuProps,
+            isOpen,
+            highlightedIndex,
+            selectItemAtIndex,
+            getLabelProps,
+          }) => {
+            return (
+              <div>
```

Comment:
Restructure this return statement into multiple methods that you would invoke here so that it can be easier to read. It's pretty long, and having the conditions is hard to read.

Reply:
created some separated render methods. thank you!